packages/headless/src/modules/customer/services/index.ts

```diff
@@ -44,20 +44,37 @@ export class CustomerService {
     }
 
     async addCustomerNewAddress(customerId: string, address: CustomerAddress): Promise<AddCustomerNewAddressResponse> {
-        let customer = await this.customerRepo.findCustomer({ id: customerId });
-        if (!customer) return this.helper.serviceResponse.errorResponse(GetCustomerInformationErrorMessages.CUSTOMER_NOT_FOUND, null, HttpStatus.BAD_REQUEST);
-
         const updatedCustomer = await this.customerRepo.addCustomerNewAddress(customerId, address);
         if (!updatedCustomer) return this.helper.serviceResponse.errorResponse(AddCustomerNewAddressErrorMessages.CAN_NOT_ADD_CUSTOMER_NEW_ADDRESS, null, HttpStatus.BAD_REQUEST);
+
+        if (address.isDefault && updatedCustomer.addresses.length && updatedCustomer.addresses.length > 1) {
+            let check = false;
+            updatedCustomer.addresses.forEach(address => {
+                if (address.isDefault && updatedCustomer.addresses[updatedCustomer.addresses.length - 1].id !== address.id) {
+                    address.isDefault = false;
+                    check = true;
+                }
+            })
+            check && await this.customerRepo.updateCustomer(customerId, updatedCustomer);
+        }
```

Comment:
seems okay. but i think you could do it in more clean way. you could write a repository function like setDefaultAddress(defaultAddressId: string){
  make all the addresses isDefault: false;
set isDefault: true for the defaultAddressId.
}